packages/components/src/DateTimePickers/InputDateTimePicker/InputDateTimePicker.component.js

```diff
@@ -0,0 +1,221 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+
+import getMinutes from 'date-fns/get_minutes';
+import getHours from 'date-fns/get_hours';
+import getDate from 'date-fns/get_date';
+import getMonth from 'date-fns/get_month';
+import getYear from 'date-fns/get_year';
+import setDate from 'date-fns/set_date';
+import setMinutes from 'date-fns/set_minutes';
+import lastDayOfMonth from 'date-fns/last_day_of_month';
+import { twoDigits } from '../pickers/TimePicker';
+import DateTimePicker from '../DateTimePicker';
+import theme from './InputDateTimePicker.scss';
+
+const splitDateAndTimeRegex = new RegExp(/^(.*),(.*)$/);
+const dateRegex = new RegExp(/^\s*([0-9]{1,2})\/([0-9]{1,2})\/([0-9]{4})\s*$/);
+const timeRegex = new RegExp(/^\s*([0-9]{1,2}):([0-9]{2})\s*$/);
+
+function hoursAndMinutesToTime(hours, minutes) {
+	return hours * 60 + minutes;
+}
+
+function getTextDate(date, time) {
+	if (date === undefined) {
+		return '';
+	}
+
+	const day = getDate(date);
+	const month = getMonth(date) + 1;
+	const year = getYear(date);
+
+	const dateText = `${twoDigits(day)}/${twoDigits(month)}/${year}`;
+
+	if (time === undefined) {
+		return dateText;
+	}
+
+	const hours = Math.floor(time / 60);
+	const minutes = time % 60;
+
+	const timeText = `, ${twoDigits(hours)}:${twoDigits(minutes)}`;
+
+	return dateText + timeText;
+}
+
+
+class InputDateTimePicker extends React.Component {
+	static propTypes = {
+		selectedDateTime: PropTypes.instanceOf(Date),
+		onChange: PropTypes.func.isRequired,
+	};
+
+	constructor(props) {
+		super(props);
+
+		this.state = (() => {
+			const selectedDateTime = this.props.selectedDateTime;
+			if (selectedDateTime !== undefined) {
+				const date = selectedDateTime;
+				const hours = getHours(selectedDateTime);
+				const minutes = getMinutes(selectedDateTime);
+				const time = hoursAndMinutesToTime(hours, minutes);
+				return {
+					date,
+					time,
+					textInput: getTextDate(date, time),
+				};
+			}
+
+			return {
+				textInput: '',
+			};
+		})();
+
+		this.onChangeInput = this.onChangeInput.bind(this);
+		this.onSubmitPicker = this.onSubmitPicker.bind(this);
+	}
+
+	onSubmitPicker({ date, time }) {
+		this.updateDateTime(date, time);
+	}
+
+	onChangeInput(event) {
+		const text = event.target.value;
+
+		const splitMatches = text.match(splitDateAndTimeRegex);
+
+		if (!splitMatches) {
+			console.warn('DATETIME - INCORRECT FORMAT:', text);
+			this.updateDateTime(undefined, undefined, text);
+			return;
+		}
+
+		const [
+			,
+			dateText,
+			timeText,
+		] = splitMatches;
+
+		const date = (() => {
+			const dateMatches = dateText.match(dateRegex);
+			if (!dateMatches) {
+				console.warn('DATE - INCORRECT FORMAT:', dateText);
+				return undefined;
+			}
+
+			const [
+				,
+				dayString,
+				monthString,
+				yearString,
+			] = dateMatches;
+
+			const day = parseInt(dayString, 10);
+			const month = parseInt(monthString, 10);
+			const monthIndex = month - 1;
```

Comment:
You can get rid of `month`

Reply:
I can get rid of so many things... but does it gain something apart from one line, to do all in one line ?
I find it much simpler with one instruction at a time and more meaningful why we are decrementing of one.

Reply:
Making it inline is fully understandable without any problem. So creating a new var for something that simple doesn't help.

Reply:
But it doesn't hurt too ...
It's much easier to reason globally when each value is hold by one variable. Otherwise it makes it totally subjective when to break it in multiple lines or keep it in one line. Where are the objective limits in both way (too long and too short) ?

Reply:
I use it later so I keep it like this.
✘